tests/integration/test_insert_into_distributed/test.py

```diff
@@ -75,6 +75,18 @@ def started_cluster():
         node2.query('''
 CREATE TABLE table_function (n UInt8, s String) ENGINE = MergeTree() ORDER BY n''')
 
+        node1.query('''
+CREATE TABLE distributed_one_replica (date Date, id UInt32) ENGINE = Distributed('shard_with_local_replica', 'default', 'single_replicated')
+''')
+
+        node2.query('''
+CREATE TABLE distributed_one_replica (date Date, id UInt32) ENGINE = Distributed('shard_with_local_replica', 'default', 'single_replicated')
+''')
+
+        node2.query('''
+CREATE TABLE single_replicated(date Date, id UInt32) ENGINE = ReplicatedMergeTree('/clickhouse/tables/0/single_replicated', 'node2', date, id, 8192)
```

Comment:
You are checking for distributed table not underlying table (like skip_unavaialble_shards assumes) so here node1 should not have distributed table too

Also looks like it worth updating documentation for skip_unavailable_shards to add information that it will skip the remote shard if table does not exists.

Reply:
Oops, I just realize that I have made a wrong title. It's to skip unavailable replicas.

Reply:
@amosbird please fix test, other changes look good to me. @azat are there any concerns with current implementation ? I checked and read implemented similarly.